cpp/src/request.cpp

```diff
@@ -69,7 +69,7 @@ void Request::cancel()
                        ucs_status_string(status));
     } else {
       ucxx_trace_req_f(_ownerString.c_str(), _request, _operationName.c_str(), "canceling");
-      ucp_request_cancel(_worker->getHandle(), _request);
+      if (_request != nullptr) ucp_request_cancel(_worker->getHandle(), _request);
```

Comment:
This is unrelated to the AM changes and just a check we need to add, right?

Reply:
That's right, I found it during development here and decided to throw it in here instead of opening a new PR, although I'll probably add this to another PR too to get this change in sooner.